src/typing/typer.ml

```diff
@@ -1571,13 +1585,20 @@ let type_bind ctx (e : texpr) (args,ret) params p =
 		| [], [] -> given_args,missing_args,ordered_args
 		| [], _ -> error "Too many callback arguments" p
 		| (n,o,t) :: args , [] when o ->
-			let a = if is_pos_infos t then
+			let a = begin match follow t with
+			| TAbstract({a_impl = Some c; a_from_nothing = Some cf} as a,tl) ->
+				let t = AbstractCast.make_static_call ctx c cf a tl [] t p in
+				ordered_args @ [t]
+			| _ ->
+				if is_pos_infos t then
 					let infos = mk_infos ctx p [] in
 					ordered_args @ [type_expr ctx infos (WithType t)]
 				else if ctx.com.config.pf_pad_nulls then
 					(ordered_args @ [(mk (TConst TNull) t_dynamic p)])
 				else
-					ordered_args
+					ordered_args @ [(mk (TConst TNull) t_dynamic p)]
```

Comment:
@ncannasse does this change, actually ignoring pf_pad_null for bind, does have any effects on real code? The unit tests still they pass.

Reply:
well it's not ignoring pf_pad_null but is always pushing nulls whereas you're padding or not. I'm not sure I like it, but I guess we can remove extra nulls at a later stage maybe?

Reply:
yes, true. I actually wonder if and what implications this actually have.

Reply:
or to be more precise, which negative implications this have.

Reply:
and just to let you know, there are already cases where this leads to bugs: http://try-haxe.mrcdk.com/#12d04.